moto/sagemaker/models.py

```diff
@@ -3397,15 +3405,32 @@ def create_model_package(
         additional_inference_specifications: Any,
     ) -> str:
         model_package_version = None
-        if model_package_group_name is not None:
+        if model_package_group_name and model_package_name:
+            raise AWSValidationException(
+                "An error occurred (ValidationException) when calling the CreateModelPackage operation: Both ModelPackageName and ModelPackageGroupName are provided in the input. Cannot determine which one to use."
+            )
+        elif not model_package_group_name and not model_package_name:
+            raise AWSValidationException(
+                "An error ocurred (ValidationException) when calling the CreateModelPackag operation: Missing ARN."
```

Comment:
This  is needed since `ModelPackageName` and `ModelPackageGroupName` are `Optional`. Unfortunately, AWS response in this case is pretty general "Missing ARN". (probably referring to any one of these fields).